### PR TITLE
[CutBranching] New wrapper for maximum independent set 

### PR DESCRIPTION
**Pull request by claud10cv**
This PR creates a Julia wrapper for the CutBranching C++ package for the maximum independent set (Demian Hespe et al, Targeted Branching for the Maximum Independent Set Problem, 19th International Symposium on Experimental Algorithms (SEA 2021)). The upstream code is available at https://github.com/Hespian/CutBranching. Minor modifications (including a Julia connector) are included in my own fork of the code from which thir PR originates.

### CODE REVIEW

File: C/CutBranching/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+cmake -B build -DCMAKE_INSTALL_PREFIX=${prefix} -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DJulia_PREFIX=${Julia_PREFIX} -DJlCxx_DIR=$prefix/lib/cmake/JlCxx
+cmake --build build --parallel ${nproc}
+cmake --install build
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+filter!(x -> libc(x) != "musl", platforms)
+filter!(x -> os(x) != "windows", platforms)
```
Comment (giordano) [inline C/CutBranching/build_tarballs.jl:26]:
```suggestion
filter!(!Sys.iswindows, platforms)
```

File: C/CutBranching/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+#platforms = [
+#    Platform("x86_64", "linux"; libc = "glibc"),
+#]
+
+
+# The products that we will ensure are always built
+products = Product[
+	LibraryProduct(["libmiscb", "miscblib"], :miscb)
```
Comment (giordano) [inline C/CutBranching/build_tarballs.jl:34]:
```suggestion
products = [
    LibraryProduct(["libmiscb", "miscblib"], :miscb)
```

File: C/CutBranching/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+	Dependency("libcxxwrap_julia_jll"),
+	Dependency("libjulia_jll")
+
```
Comment (giordano) [inline C/CutBranching/build_tarballs.jl:41]:
```suggestion
dependencies = [
    Dependency("libcxxwrap_julia_jll"),
    Dependency("libjulia_jll")
```

**Review by giordano — COMMENTED** [review]
Again, for packages linking to libjulia follow existing examples: https://github.com/JuliaPackaging/Yggdrasil/pull/8337#pullrequestreview-1953535572

**Comment by claud10cv** [comment]
Thank you @giordano, I was updating the file while you were proposing new changes. I updated the file accordingly

**Comment by claud10cv** [comment]
Hello @giordano, Is there any other review/comment that I should integrate before this can be scheduled for merge?

**Pull request merged**